terraform/ec2/linux/main.tf

```diff
@@ -26,6 +26,51 @@ locals {
   private_key_content = var.ssh_key_name != "" ? var.ssh_key_value : tls_private_key.ssh_key[0].private_key_pem
 }
 
+#####################################################################
+# Create EFS
+#####################################################################
+resource "aws_efs_file_system" "efs" {
+  creation_token = "efs-${random_id.testing_id.hex}"
+  tags           = {
+    Name = "efs-${random_id.testing_id.hex}"
+  }
+}
+
+resource "aws_efs_mount_target" "mount" {
+  file_system_id = aws_efs_file_system.efs.id
+  subnet_id = aws_instance.cwagent.subnet_id
+  security_groups = [aws_security_group.ec2_security_group.id]
+}
+
+resource "null_resource" "mount_efs" {
```

Comment:
Do we need the null resource? Should we add the depends on what the null resource depends on instead of creating a null resource.

Reply:
> Should we add the depends on what the null resource depends on instead of creating a null resource.

Could you break down what this means?